### PR TITLE
WT-7091 Restrict LSM tests with compatible backup configuration

### PR DESCRIPTION
**Pull request by raviprakashgiri29**

### CODE REVIEW

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by agorrod** [comment]
@raviprakashgiri29 would you mind updating our backup documentation to reflect this limitation? Bonus points for restricting the backup API usage if LSM trees are present and reporting a helpful error message.

**Comment by raviprakashgiri29** [comment]
> @raviprakashgiri29 would you mind updating our backup documentation to reflect this limitation? Bonus points for restricting the backup API usage if LSM trees are present and reporting a helpful error message.

Thanks, @agorrod for pointing this. Updated the documentation and added a condition to return an error when backup API is used with open LSM trees.

**Comment by sueloverso** [comment]
These changes are fine for a work-around. But I'd like a new ticket to know what is happening for LSM because off-hand it should work for LSM. We should have a bug-fixing ticket to investigate that.

File: src/cursor/cur_backup.c [hunk 1]
```diff
@@ -333,6 +333,12 @@ __backup_add_id(WT_SESSION_IMPL *session, WT_CONFIG_ITEM *cval)
 
     conn = S2C(session);
     blk = NULL;
+
+    /* Return error if block-based incremental backup is performed with open LSM trees.  */
+    if (!TAILQ_EMPTY(&conn->lsmqh)) {
+        WT_RET_MSG(session, ENOTSUP, "LSM does not work with block-based incremental backup");
+    }
```
Comment (sueloverso) [inline src/cursor/cur_backup.c:340]: I think this should move (and use the `incremental_config` setting) to the compatibility testing chunk of code in `__backup_config` in this file.

File: src/cursor/cur_backup.c [hunk 2]
```diff
@@ -333,6 +333,12 @@ __backup_add_id(WT_SESSION_IMPL *session, WT_CONFIG_ITEM *cval)
 
     conn = S2C(session);
     blk = NULL;
+
+    /* Return error if block-based incremental backup is performed with open LSM trees.  */
```
Comment (sueloverso) [inline src/cursor/cur_backup.c:337]: Minor: Return an error...

File: src/docs/backup.dox [hunk 3]
```diff
@@ -171,6 +171,10 @@ database directory has not been opened and recovery run.  Once recovery
 has run in a backup directory, you can no longer back up to that
 database directory.
 
+The only limitation of block-based incremental backup is that it does not
+work with LSM trees. It will throw an error if backup API is used to do
+block-based incremental backup with LSM trees.
```
Comment (sueloverso) [inline src/docs/backup.dox:176]: I suggest making this more of an active voice:
Block-based incremental backup does not work with LSM trees. An error will be returned in that case.

**Review by sueloverso — COMMENTED** [review]
I have one request for moving the library code and a couple minor things. Otherwise this looks good and is very close.

File: src/cursor/cur_backup.c (hunk 1 above)
Comment (raviprakashgiri29) [reply src/cursor/cur_backup.c]: Moved to check the condition to  `__backup_config ` function.

File: src/docs/backup.dox (hunk 3 above)
Comment (raviprakashgiri29) [reply src/docs/backup.dox]: Updated

File: src/cursor/cur_backup.c (hunk 2 above)
Comment (raviprakashgiri29) [reply src/cursor/cur_backup.c]: Fixed

**Comment by raviprakashgiri29** [comment]
> These changes are fine for a work-around. But I'd like a new ticket to know what is happening for LSM because off-hand it should work for LSM. We should have a bug-fixing ticket to investigate that.

Thanks, @sueloverso for pointing this out. I created a ticket WT-7090 last week to fix these failures.

**Review by sueloverso — APPROVED** [review]
These changes LGTM. I have one request. Can you itemize, in WT-7090, the locations of all the things that need to be updated when that is fixed so that none of these places is forgotten (like the documentation).

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**